packages/cli/src/cmds/validator/voluntaryExit.ts

```diff
@@ -0,0 +1,171 @@
+import inquirer from "inquirer";
+import {computeSigningRoot, computeEpochAtSlot, computeStartSlotAtEpoch} from "@lodestar/state-transition";
+import {DOMAIN_VOLUNTARY_EXIT} from "@lodestar/params";
+import {createIBeaconConfig} from "@lodestar/config";
+import {ssz, phase0} from "@lodestar/types";
+import {toHex} from "@lodestar/utils";
+import {Signer, SignerLocal, SignerType} from "@lodestar/validator";
+import {Api, getClient} from "@lodestar/api";
+import {ensure0xPrefix, ICliCommand, YargsError} from "../../util/index.js";
+import {IGlobalArgs} from "../../options/index.js";
+import {getBeaconConfigFromArgs} from "../../config/index.js";
+import {IValidatorCliArgs} from "./options.js";
+import {getSignersFromArgs} from "./signers/index.js";
+
+/* eslint-disable no-console */
+
+type VoluntaryExitArgs = {
+  exitEpoch?: number;
+  pubkeys?: string[];
+  yes?: boolean;
+};
+
+export const voluntaryExit: ICliCommand<VoluntaryExitArgs, IValidatorCliArgs & IGlobalArgs> = {
+  command: "voluntary-exit",
+
+  describe:
+    "Performs a voluntary exit for a given validator (as identified via `publicKey`.  \
+If no `publicKey` is provided, a prompt will ask the user which validator they would \
+like to choose for the voluntary exit.",
+
+  examples: [
+    {
+      command: "account validator voluntary-exit --publicKey 0xF00",
+      description: "Perform a voluntary exit for the validator who has a public key 0xF00",
+    },
+  ],
+
+  options: {
+    exitEpoch: {
+      description:
+        "The epoch upon which to submit the voluntary exit.  If no value is provided, then we default to the currentEpoch.",
+      type: "number",
+    },
+
+    pubkeys: {
+      description: "Pubkeys to exit, must be available as local signers",
+      type: "array",
+      string: true, // Ensures the pubkey string is not automatically converted to numbers
+      coerce: (pubkeys: string[]): string[] =>
+        // Parse ["0x11,0x22"] to ["0x11", "0x22"]
+        pubkeys
+          .map((item) => item.split(","))
+          .flat(1)
+          .map(ensure0xPrefix),
+    },
+
+    yes: {
+      description: "Skip confirmation prompt",
+      type: "boolean",
+    },
+  },
+
+  handler: async (args) => {
+    const network = args.network;
+
+    // Fetch genesisValidatorsRoot always from beacon node
+    // Do not use known networks cache, it defaults to mainnet for devnets
+    const chainForkConfig = getBeaconConfigFromArgs(args);
+    const client = getClient({baseUrl: args.server}, {config: chainForkConfig});
+    const {genesisValidatorsRoot} = (await client.beacon.getGenesis()).data;
+    const config = createIBeaconConfig(chainForkConfig, genesisValidatorsRoot);
+
+    // Fetch exitEpoch from latest header if unspecified
+    const exitEpoch =
+      args.exitEpoch ?? computeEpochAtSlot((await client.beacon.getBlockHeader("head")).data.header.message.slot);
```

Comment:
Doesn't practically matter, but we can use the clock slot (using above genesis data) instead of the head block slot. Then we don't need an additional api call.

Reply:
Good idea